lib/FetchEnvs.js

```diff
@@ -267,6 +289,27 @@ function reduceEnv(output, conf) {
   return output;
 }
 
+function reduceEnvSourceSimpleLinks(output, conf) {
+  const { valueFrom, parentNamespace  } = conf;
+  if (valueFrom !== undefined) {
+    let sourceValueFromInfo = Object.values(valueFrom)[0];
```

Comment:
How do we know that the key we want will always be first? Seems like this could be unpredictable later. I'd suggest use a well know key name to get the value out of the Object that you want

Reply:
From what I understand valueFrom is always in the format 
```
{
 "genericMapRef": {
  "apiVersion": "deploy.razee.io/v1alpha2",
  "kind": "FeatureFlagSetLD",
  "name": "test-dummy-ffsld",
  "namespace": "test-namespace"
 }
}

```
so we could move the `Object.keys(valueFrom)[0]` that gets the ref type up from where it is in the else{KIND_MAP} part of things and use that as the key to get the values instead of Object.values(valueFrom)[0].
That should work since valueFrom is always for a single ref, but it doesn't get rid of the future uncertainty issue

Reply:
ie valuefrom is always an object with just 1 key, but we don't know which type of ref it will be?

Reply:
Correct

Reply:
More specifically it will be whatever type of ref the source is from (i.e. ConfigMapRef, GenericMapRef, ConfigMapKeyRef, etc.)